### PR TITLE
wpewebkit: Enable QtWPE build support for Qt6 for 2.46.0

### PR DESCRIPTION
**Pull request by Qingwu-Li**
Tested with QT6 build

know issue：
gcc 11 may failed build [neon.h](https://github.com/WebKit/WebKit/blob/main/Source/WTF/wtf/simde/arm/neon.h) failed for arm64, see https://github.com/Qingwu-Li/WebKit/commit/176cb80e1579308f5f4a534ee796b1a7855f6248

### CODE REVIEW

**Comment by Qingwu-Li** [comment]
@pabelenda Please do a review!

**Comment by psaavedra** [comment]
Hi @Qingwu-Li, thank you for the PR. We really appreciate all contributions to the project. I have a few comments regarding the pull request:

- `2.45.x` is not a stable WPE release. Odd-numbered versions refer to development releases. I suggest adjusting the `wpewebkit` bump to version `2.44.4`. Please note that [this PR](https://github.com/Igalia/meta-webkit/pull/505) already handles the bump to `2.44.4`.
- Is Qt6 WPE not yet supported in `2.44`? If that is the case, we could consider adding support for `2.45` as an exception, but with `DEFAULT_PREFERENCE = "-1"` in the recipe.

**Comment by Qingwu-Li** [comment]
> * Is Qt6 WPE not yet supported in `2.44`? If that is the case, we could consider adding support for `2.45` as an exception, but with `DEFAULT_PREFERENCE = "-1"` in the recipe.

Hello @psaavedra , yes Qt6 WPE not supported in 2.44,
Change in V2:
 Add  `DEFAULT_PREFERENCE = "-1"`
Fix the gcc 11(kirkstone) build simd failed issue.

**Comment by aperezdc** [comment]
We are going to release 2.46.0 soon. It would be great if you could update this PR with an update to that version when it's available in some weeks.

**Comment by psaavedra** [comment]
> We are going to release 2.46.0 soon. It would be great if you could update this PR with an update to that version when it's available in some weeks.

This PR (https://github.com/Igalia/meta-webkit/pull/509) will move the WPE version to 2.46. I could be very happy to integrate your changes after this 509 is merged if you make the adaptations in this 506 PR.

**Comment by Qingwu-Li** [comment]
Got 2 issue during compile:
For 'wpe_view_resize' was not declared, change to wpe_view_resized fixed

but for 'wpe_toplevel_qtquick' has not been declared, how to fix?
## 1  'wpe_view_resize' was not declared
```
Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:87:5: error: 'wpe_view_resize' was not declared in this scope; did you mean 'wpe_view_resized'?
|    87 |     wpe_view_resize(wpeView, d->m_size.width(), d->m_size.height());
|       |     ^~~~~~~~~~~~~~~
|       |     wpe_view_resized
```
Modify `wpe_view_resize `to `wpe_view_resized `fixed

## 2  'wpe_toplevel_qtquick' has not been declared

```
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:35:46: error: 'wpe_toplevel_qtquick' has not been declared
|    35 | WEBKIT_DEFINE_FINAL_TYPE(WPEToplevelQtQuick, wpe_toplevel_qtquick, WPE_TYPE_TOPLEVEL, WPEToplevel)
|       |                                              ^~~~~~~~~~~~~~~~~~~~
| In file included from Source/WebKit/WPEPlatform/wpe/WPEView.h:36,
|                  from Source/WebKit/WPEPlatform/wpe/WPEEvent.h:35,
|                  from Source/WebKit/WPEPlatform/wpe/wpe-platform.h:31,
|                  from Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.h:30,
|                  from Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.h:29,
|                  from Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:27:
| Source/WebKit/WPEPlatform/wpe/WPEToplevel.h:38:27: error: expected identifier before '(' token
|    38 | #define WPE_TYPE_TOPLEVEL (wpe_toplevel_get_type())
|       |                           ^
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:35:68: note: in expansion of macro 'WPE_TYPE_TOPLEVEL'
|    35 | WEBKIT_DEFINE_FINAL_TYPE(WPEToplevelQtQuick, wpe_toplevel_qtquick, WPE_TYPE_TOPLEVEL, WPEToplevel)
|       |                                                                    ^~~~~~~~~~~~~~~~~
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:37:1: error: expected constructor, destructor, or type conversion before 'static'
|    37 | static gboolean wpeToplevelQtQuickResize(WPEToplevel* toplevel, int width, int height)
|       | ^~~~~~
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp: In function 'void wpe_toplevel_qtquick_class_init(WPEToplevelQtQuickClass*)':
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:52:29: error: 'wpeToplevelQtQuickResize' was not declared in this scope; did you mean 'WPEToplevelQtQuickClass'?
|    52 |     toplevelClass->resize = wpeToplevelQtQuickResize;
|       |                             ^~~~~~~~~~~~~~~~~~~~~~~~
|       |                             WPEToplevelQtQuickClass
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp: At global scope:
| Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:49:13: warning: 'void wpe_toplevel_qtquick_class_init(WPEToplevelQtQuickClass*)' defined but not used [-Wunused-function]
|    49 | static void wpe_toplevel_qtquick_class_init(WPEToplevelQtQuickClass* toplevelQtQuickClass)
|       |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
No idea yet!

**Comment by psaavedra** [comment]
> > We are going to release 2.46.0 soon. It would be great if you could update this PR with an update to that version when it's available in some weeks.
> 
> This PR (#509) will move the WPE version to 2.46. I could be very happy to integrate your changes after this 509 is merged if you make the adaptations in this 506 PR.

#509 is ready and landed. Let's trigger the Actions for this PR.

File: recipes-browser/wpewebkit/wpewebkit/0003-WPE-Platform-fix-wpe_toplevel_qtquick-has-not-been-d.patch [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Subject: [PATCH 3/4] WPE Platform: fix wpe_toplevel_qtquick has not been
+ declared
+
+Fix the following compile failed issue for Qt6:
+
+Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp:35:46:
+error: 'wpe_toplevel_qtquick' has not been declared
+
+Upstream-Status: Pending [1]
+[1] https://github.com/WebKit/WebKit/pull/34441
```
Comment (psaavedra) [inline recipes-browser/wpewebkit/wpewebkit/0003-WPE-Platform-fix-wpe_toplevel_qtquick-has-not-been-d.patch:13]: s/Pending/Submitted/

File: recipes-browser/wpewebkit/wpewebkit/0004-WPE-Platform-fix-wpe_view_resize-was-not-declared-is.patch [hunk 2]
```diff
@@ -0,0 +1,48 @@
+From b92e21659ae4621240e444156512961a14007881 Mon Sep 17 00:00:00 2001
+From: LI Qingwu <Qing-wu.Li@leica-geosystems.com.cn>
+Date: Thu, 10 Oct 2024 12:56:58 +0800
+Subject: [PATCH 4/4] WPE Platform: fix wpe_view_resize was not declared issue
+
+fix Source/WebKit/UIProcess/API/wpe/qt6/WPEQtView.cpp:87:5:
+error: 'wpe_view_resize' was not declared in this scope;
+did you mean 'wpe_view_resized'?
+
+Upstream-Status: Pending [1]
```
Comment (psaavedra) [inline recipes-browser/wpewebkit/wpewebkit/0004-WPE-Platform-fix-wpe_view_resize-was-not-declared-is.patch:10]: s/Pending/Submitted/

File: recipes-browser/wpewebkit/wpewebkit-qt6.inc [hunk 3]
```diff
@@ -0,0 +1,20 @@
+inherit qt6-cmake
+
+CMAKE_QT_OECONF = "\
+    -DUSE_QT6=ON \
+    -DENABLE_WPE_PLATFORM=ON \
```
[the remark this answers is not in the document]
Comment (aperezdc) [reply recipes-browser/wpewebkit/wpewebkit-qt6.inc:5]: Yes, the Qt6 integration uses WPEPlatform. In this case it's okay to rely on it even when its API is not yet considered stable becayse the Qt6 integration is also in the WebKit tree and the expectation is that we would keep it working alongside changes done to WPEPlatform.

[the remark this answers is not in the document]
Comment (Qingwu-Li) [reply recipes-browser/wpewebkit/wpewebkit-qt6.inc:5]: @aperezdc  if I set as
```
ENABLE_WPE_PLATFORM=ON
ENABLE_WPE_PLATFORM_DRM=OFF
ENABLE_WPE_PLATFORM_HEADLESS=OFF
ENABLE_WPE_PLATFORM_WAYLAND=OFF
```
I got folowing errors
```
Source/WebKit/WPEPlatform/wpe/WPEBufferSHM.cpp:29:10: fatal error: WPEEnumTypes.h: No such file or directory
|    29 | #include "WPEEnumTypes.h"
|       |          ^~~~~~~~~~~~~~~~
```

It seems need at lease set on of then to `ON`
Keep those on as default?

**Review by psaavedra — APPROVED** [review]

**Comment by psaavedra** [comment]
@Qingwu-Li  is this PR still a Draft?

**Comment by Qingwu-Li** [comment]
modifed Upstream-Status to `Upstream-Status: Submitted`

**Pull request merged**